### PR TITLE
Allow hr to cancel their pending leave requests

### PR DESCRIPTION
**Pull request by cp-pratik-k**
### Purpose
 - allow hr for cancel their pending leave request

### Summary of Changes
 - allow hr for cancel their pending leave request

### Test steps
 - run test

### Conformity
- [x] Followed [git guidelines](https://github.com/canopas/flutter-developer-roadmap/blob/main/GitGuideline.md) for creating commit messages and Pull Request guidelines.
- [x] Self-approved the PR - reviewed the PR as a reviewer and gave it self-approval if everything is ok. If not, made the required changes.
- [x] Ensured that the PR satisfies all specified requirements in the ticket, including bug fixes and new features.
- [x] Provided test steps, including steps to reproduce the issue or test the new functionality, ensuring other team members can verify the changes.
- [x] Added/Updated proper code comments to make it easy-to-understand for other developers.
- [x] Reused code (if the same code was written twice, made it common and reused it at both places).
- [x] Removed unused or commented code if not required.
- [x] Ensured proper Dart naming conventions were used for variables, classes, and methods.
- [x] Localized user-facing strings.
- [x] Included screenshots/videos of behavior changes: Provided visual evidence of any changes to UI or behavior for easier review and understanding in the PR description.
- [x] Implemented proper error handling: Ensured that the code anticipated and handled potential errors and edge cases gracefully.
- [x] Avoided introducing technical debt: If the PR introduces technical debt, created and linked appropriate tickets for future resolution.
- [x] Included relevant unit tests: Wrote unit tests that focused on testing behavior and functionality, rather than merely covering lines of code.
- [x] Ensured code was performant and scalable: Verified that the changes did not introduce performance issues or bottlenecks and could scale as needed.
- [x] Ensured comments were up-to-date and relevant to the code to describe complex logic and to add understanding for other developers.
- [x] Marked the PR as ready before submitting it for review.

### CODE REVIEW

File: lib/ui/admin/leaves/details/widget/admin_leave_details_action_button.dart [hunk 1]
```diff
@@ -26,13 +27,15 @@ class AdminLeaveDetailsActionButton extends StatelessWidget {
...
   @override
   Widget build(BuildContext context) {
+    final userStateNotifier = getIt<UserStateNotifier>();
     final localization = AppLocalizations.of(context);
-    if (status == LeaveStatus.approved &&
-        !(getIt<UserStateNotifier>().isHR && role == Role.hr)) {
+    if ((status == LeaveStatus.approved && !(userStateNotifier.isHR && role == Role.hr)) ||
+        (uid == userStateNotifier.employeeId && status == LeaveStatus.pending)) {
       return CancelButton(onTap: () {
         showAlertDialog(
```
Comment (cp-sneha-s) [inline lib/ui/admin/leaves/details/widget/admin_leave_details_action_button.dart:40]: From this, is HR can cancel another HR's leave?

**Review by cp-sneha-s — APPROVED** [review]
